src/main/java/edu/umich/ctools/esb/utils/WAPI.java

```diff
@@ -0,0 +1,213 @@
+	}
+	
+	public String getApiPrefix() {
+		return apiPrefix;
+	}
+
+	public void setApiPrefix(String apiPrefix) {
+		this.apiPrefix = apiPrefix;
+	}
+	
+	//a token must be created at the time of construction
+	//this token will allow use of the ESB APIs
+	public String buidRenewal(String key, String secret) {
+		String b64 = base64KeySecret(key, secret);
+		b64 = "Basic " + b64; 
+		return b64;
+	}
+
+	//esb calls require base 64 strings for authroization
+	public String base64KeySecret(String key, String secret) {
+		String keySecret = key + ":" + secret;
+		byte[] binaryData = keySecret.getBytes();
+		keySecret = Base64.encodeBase64String(binaryData);
+		return keySecret;
+	}
+
+	//perform ESB request. If there is an exception, return exception result response.
+	public WAPIResultWrapper doRequest(String request) throws UnirestException{
+		M_log.info("doRequest: " + request);
+		WAPIResultWrapper wrappedResult = null;
+		JSONObject jsonObject = null;
+		HttpResponse<String> response = null;
+		try{
+			response = Unirest.get(request)
+					.header(AUTHORIZATION, this.token)
+					.header("Accept", "json")
+					.asString();
+			M_log.info("Raw body: " + response.getBody());
+			M_log.info("Status: " + response.getStatus());
+			M_log.info("Status Text: " + response.getStatusText());
+			jsonObject = new JSONObject(response.getBody());
+			wrappedResult = new WAPIResultWrapper(response.getStatus(), "COMPLETED", jsonObject);
+		}
+		catch(Exception e){
+			M_log.error("Error in doRequest: " + e.getMessage());
+			wrappedResult = reportError(response.getStatus());
+		}
+		return wrappedResult;
+	}	
+	
+	//Error handling for bad calls
+	public WAPIResultWrapper reportError(int status) {
+		M_log.info("reportError() called");
+		M_log.info("status: " + status);
+		WAPIResultWrapper wrappedResult = null;
+		switch(status){
+			
+		case HTTP_BAD_REQUEST:
+			wrappedResult = new WAPIResultWrapper(HTTP_BAD_REQUEST, "BAD REQUEST", new JSONObject("{error : " + ERROR_MSG + "}"));
+			break;
+			
+		case HTTP_UNAUTHORIZED:
+			wrappedResult = new WAPIResultWrapper(HTTP_UNAUTHORIZED, "UNAUTHORIZED", new JSONObject("{error : " + ERROR_MSG + "}"));
+			break;
+		
+		case HTTP_FORBIDDEN:
+			wrappedResult = new WAPIResultWrapper(HTTP_FORBIDDEN, "FORBIDDEN", new JSONObject("{error : " + ERROR_MSG + "}"));
+			break;	
+			
+		case HTTP_NOT_FOUND:
+			wrappedResult = new WAPIResultWrapper(HTTP_NOT_FOUND, "NOT FOUND", new JSONObject("{error : " + ERROR_MSG + "}"));
+			break;
+			
+		case HTTP_GATEWAY_TMEOUT:
+			wrappedResult = new WAPIResultWrapper(HTTP_GATEWAY_TMEOUT, "GATEWAY TIME OUT", new JSONObject("{error : " + ERROR_MSG + "}"));
+			break;
+			
+		case HTTP_UNKNOWN_ERROR:
+			wrappedResult = new WAPIResultWrapper(HTTP_UNKNOWN_ERROR, "UNKNOWN ERROR", new JSONObject("{error : " + ERROR_MSG + "}"));
+			break;
+			
+		default:
+			wrappedResult = new WAPIResultWrapper(HTTP_UNKNOWN_ERROR, "UNKNOWN ERROR", new JSONObject("{error : " + ERROR_MSG + "}"));
+		}
```

Comment:
For readability you could do  this....

In case statement 
``` 
String errorMsg=null;
switch(status){
 case HTTP_BAD_REQUEST:
             errorMsg=BAD REQUEST;
  	         break;
 case HTTP_UNAUTHORIZED:
             errorMsg=UNAUTHORIZED;
  	         break;
}
at the end you could simply return one newWAPIResultWrapper instead of repeated.
return new WAPIResultWrapper(status, errorMsg, new JSONObject("{error : " + ERROR_MSG + "}"));
```